packages/salesforcedx-utils-vscode/src/config/configUtil.ts

```diff
@@ -144,6 +146,28 @@ export class ConfigUtil {
     const username = await getUsernameFor(defaultDevHubUsernameOrAlias);
     return username ? String(username) : undefined;
   }
+
+  public static async setDefaultUsernameOrAlias(usernameOrAlias: string): Promise<void> {
+    const originalDirectory = process.cwd();
+    // In order to correctly setup Config, the process directory needs to be set to the current workspace directory
+    const workspacePath = workspaceUtils.getRootWorkspacePath();
+    try {
+      if (usernameOrAlias) { // check if username is an empty string
+        await Org.create({ aliasOrUsername: usernameOrAlias }); // check if username is valid
+      }
```

Comment:
nit: I could see pulling the above logic out into a `isUsernameValid` function.  It's a little concerning now that it's not very clear what happens if it is invalid.

Reply:
Ritam and I dug into this together and found that this isn't validating the username, but actually verifying there is a valid connection.  She's going to update the comment to more clearly state what is being checked and leave the function as is.

Reply:
`Org.create()`creates a connection with the active usernameOrAlias and throws an error if the usernameOrAlias is expired. Will update the comment for it to make more sense.